cmd/linkcheck/links.go

```diff
@@ -67,8 +67,8 @@ var (
 	}
 
 	visitedURLs    = map[string]struct{}{}
-	htmlpreviewReg = regexp.MustCompile(`https://htmlpreview\.github\.io/\?`)
-	httpOrhttpsReg = regexp.MustCompile(`https?.*`)
+	htmlpreviewReg = regexp.MustCompile(`^https://htmlpreview\.github\.io/\?`)
+	httpOrhttpsReg = regexp.MustCompile(`^https?.*`)
```

Comment:
I wonder if in this case we do want to catch typos where somebody may have mistyped a url, such as "hhttps"?

I think the other entries are ok to have the stricter check.

Reply:
Hi @msau42 , thanks for your comment, I agree with you. So this is the final conclusion that we should use the stricter check, right

Reply:
I meant specifically for `httpOrhttpsReg`, since the logic that uses it will [skip the string and continue](https://github.com/kubernetes/kubernetes/blob/9d3f14ab7d149c573cd365f07b7c9e03dba42c75/cmd/linkcheck/links.go#L95), I wonder if we should actually be *less* strict so that we can error on typos.

Reply:
IMHO, we should not tolerate typo, which is abnormal situation 😅

Reply:
@msau42 Any other comments?